android/WALT/app/src/main/java/org/chromium/latency/walt/ScreenResponseFragment.java

```diff
@@ -410,9 +429,42 @@ public void run() {
             blackBox.setBackgroundColor(color_gray);
             isTestRunning = false;
             startButton.setEnabled(true);
+            drawBrightnessChart();
         }
     };
 
+    private void drawBrightnessChart() {
+        final String brightnessCurveString = brightnessCurveData.toString();
+        List<Entry> entries = new ArrayList<>();
+
+        // "u" marks the start of the brightness curve data
+        int startIndex = brightnessCurveString.indexOf("u") + 1;
+        int endIndex = brightnessCurveString.indexOf("end");
+        if (endIndex == -1) endIndex = brightnessCurveString.length();
+
+        String[] brightnessStrings =
+                brightnessCurveString.substring(startIndex, endIndex).trim().split("\n");
+        for (String str : brightnessStrings) {
+            String[] arr = str.split(" ");
+            entries.add(new Entry(Integer.parseInt(arr[0]), Integer.parseInt(arr[1])));
+        }
+        LineDataSet dataSet = new LineDataSet(entries, "Brightness");
+        dataSet.setColor(Color.BLACK);
+        dataSet.setValueTextColor(Color.BLACK);
+        dataSet.setCircleColor(Color.BLACK);
+        dataSet.setCircleRadius(1.5f);
+        dataSet.setCircleColorHole(Color.DKGRAY);
+        LineData lineData = new LineData(dataSet);
+        chart.setData(lineData);
+        final Description desc = new Description();
+        desc.setText("Screen Brightness (0-1024) vs. Time (µs)");
```

Comment:
Let's change this to:
Screen Brightness [digital level 0-1023] vs. Time [ms] 
There is no clear convention on how to denote units on graphs but in all previous documentation related to WALT I used square brackets.